### ISSUE TITLE
Build 0.9.1 ?

### ISSUE THREAD

**Issue by haakonstorm** [KittyNighthawk/binary-cookie-extractor#1]
Nice tool! Newbie with `golang` here; I'm having issue building the latest due to 'tags'; how to build or go get the latest version?
I'm looking to get a one-liner to extract the cookies for Notion.so to use the unofficial API.

Your `JSON` output is killer. 

For now, I'm using `binarycookies` to get to the cookies, but it has no structured data output. 
My `grep` + `awk` + `sed` plumbing would be better replaced by a proper `jq` parse.

**Comment by KittyNighthawk** [comment]
Hiya. Thank you for reporting this, I will investigate this soon and see what the problem is 🙂

And from what I can gather, you'd like a JSON output format for cookies? I think I can do that, and it does sound like a handy feature to add.

I will update this ticket as I learn more.

And thank you for using the tool! 😊

**Comment by haakonstorm** [comment]
Wow, that was a quick and cool response for sure, thanks a bunch for that! And yes, exactly, assume `binary-cookie-extractor` can read and parse the .binarycookie format, there's [potentially [multiple] embedded] binary plist data there available for the reading. Those can all be embedded in a monolith JSON for export, and `binary-cookie-extractor` does *exactly* what it is made to do; no more no less; thus adhering to the unix philosophy.

If I might add to this request: 
imagine using some sort of generic-purpose selector utility (like `fzf`), and building everything that you need, for whatever it is that your are building - _purely_ as a shell pipeline. How could we model controls for `b-c-e` to make that pipeline as efficient as possible? 

Well, for once, Apple made us a monolith `Cookies.binarycookies`. 

So, somehow we would need to i) list, ii) search, and iii) select, from what `b-c-e` can extract for us. Perhaps a way to plug in to `fzf` directly, having this in mind? Check out the `--preview` option for `fzf`; perhaps we can somehow make a clever way to collapse the three steps of listing, searching, selecting to fewer steps? 

For this particular use case (`Notion.so` unofficial API; token extraction) I know that what I'm looking for is the value for the `notion_browser_id` key:

- If this value can be returned "bare", both with or without quotation, somebody or some script might save a few seconds here and there on each iteration? 
- Perhaps first run just extracts the key id's for selection, then using `fzf --multi` we get extracted only the key/values we selected? 
- Perhaps they can be set and exported directly as environment variables? Often that is where they end up. And, on that note, check out the japanese `envchain` project. I've started leveraging it for all my shell needs. Basically it stuffs your secrets in your operating systems keychain, transparently, using Keychain/`security cli` (macos) or gnu/gnome keyring (linux) via very simple but very easy to use user-defined namespaces.
- It's my view that tools that make such pipelines easy to set up, be transparent, _fast_, natively binary compile-able (like your tool! blazing fast! written in golang! delicious open sauce! excellent developer feedback!) that tend to win, and man, its so nice to be in and benefit from and contribute to - our comminty of sauce chefs, 'raite? :-)

**Comment by haakonstorm** [comment]
Now when I'm already ranting 🤣 - give this a spin with the thinker:

- Check out very popular extension "Sidebery" for Firefox / "Vertical Tree Tabs" for Chrome-sies. 
- Now google "Vertical tabs" and "Safari"; millions want this for Firefox; none exist
- Open `Safari` (fact: the fastest browser on macOS) -- enable `View >> Enable Sidebar` and collapse everything so that  you are at 'root' of all your bookmarjs -- wow, sure looks like a good starting point for a Vertical Tabs browsing experience, huh?
- Now examine `~/Library/Bookmarks.plist`. Try to use stock `plutil` tools to convert to JSON ... that will fail. 

To illustrate:
<img width="1436" alt="Screenshot of iTerm2 (16-01-2021, 23-03-40)" src="https://user-images.githubusercontent.com/193781/104824085-3df41880-584f-11eb-81af-3890eb2cab07.png">

```
cat ~/Library/Safari/Bookmarks.plist |
  plutil -convert xml1 -o - -- -|
  plist-to-json /dev/stdin|
  jq '.[].Children[] | select(.Title=="🌝")'
```

I had to resort to a JS tool, `plist-to-json`, to properly convert to json, but you get the idea?
By simply file-watching `Bookmarks.plist`, and some means to read and write URLs to it, we can get a _vertical_ tree-tabs style feature, simply by leveraging the shell. Vertical, tree-style, tabs, in browsers, are awesome. Safari should have them, too, right? 

I declare us `awesome` iff we can write [a one-liner | something] that use `brew services add` to leverage macOS process number 1, `launchd`, that already has file-watching functionality, to read and write to `Bookmarks.plist`, enabling vertical tree-style tabs functionality. I sincerely think we can do it with one line of code; and spawn a pretty efficient *daemon* round-tripping via your `binary-cookie-extractor` facilitate to enable it all. 🤣

**Comment by haakonstorm** [comment]
(So you see where the moon face 🌝 fit in to it all:)

<img width="1012" alt="Screenshot of Safari (16-01-2021, 23-14-58)" src="https://user-images.githubusercontent.com/193781/104824279-aabbe280-5850-11eb-9aed-23ec951e0465.png">

**Comment by KittyNighthawk** [comment]
Lot's to read in your comments!

For now though, I have added the ability to output decoded cookies in JSON, XML, and CSV formats (commit dc2a7be8b4d7c2f7beabcce56bbedbc46d39a566). I also did a little refactoring just to make it easier if I ever add any more functionality.

I couldn't get the tags error to appear when I built from a clean installation of Go. It didn't really make sense though as that version (0.9) wasn't using any struct tags. It is now though to support JSON and XML. Maybe the app generating the binary cookies file is doing something to the default format?

For reference, I installed by getting the latest Go package from https://golang.org, then running:

```$ go get github.com/KittyNighthawk/binary-cookie-extractor```

This then got the source, and built the binary (placing it in the go directory within the user's home directory). Once the home directory was added to the local ```PATH``` environment variable, everything works nice.

Hopefully, this helps with your requirement of the tool 🙂
